src/librustdoc/clean/cfg.rs

```diff
@@ -171,10 +171,15 @@ impl Cfg {
 
     /// Renders the configuration for long display, as a long HTML description.
     pub(crate) fn render_long_html(&self) -> String {
-        let on = if self.should_use_with_in_description() { "with" } else { "on" };
+        let on = if self.omit_preposition() {
```

Comment:
The `render_long_plain`  function below should also have this change applied. Really it seems like we could extract a shared function that they both delegate to since the only difference is the format AFAICT.

Reply:
Good point, sending a PR.